Failure to compile with `--cfg tokio_unstable` and the `process` and `rt` features enabled

If using the unstable features with only the `process` and `rt` features enabled, tokio will fail to compile:
```
noah@resf-attack-drone ‹ master ●●● › : ~/CLionProjects/smoke
[0] % cargo check
    Checking tokio v1.18.2
error[E0432]: unresolved import `crate::runtime::IoDriverMetrics`
  --> /home/noah/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/io/driver/metrics.rs:20:24
   |
20 |         pub(crate) use crate::runtime::IoDriverMetrics;
   |                        ^^^^^^^^^^^^^^^^---------------
   |                        |               |
   |                        |               help: a similar name exists in the module: `WorkerMetrics`
   |                        no `IoDriverMetrics` in `runtime`

error[E0282]: type annotations needed
   --> /home/noah/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/io/driver/mod.rs:303:18
    |
303 |             self.inner().map(|inner| f(&inner.metrics))
    |                  ^^^^^ cannot infer type
    |
    = note: type must be known at this point

error[E0599]: no method named `inner` found for reference `&io::driver::Handle` in the current scope
   --> /home/noah/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/io/driver/mod.rs:303:18
    |
303 |             self.inner().map(|inner| f(&inner.metrics))
    |                  ^^^^^-- help: remove the arguments
    |                  |
    |                  field, not a method

Some errors have detailed explanations: E0282, E0432, E0599.
For more information about an error, try `rustc --explain E0282`.
error: could not compile `tokio` due to 3 previous errors
```

I ran a bisect and found it was introduced here:
```
769fb1547fc50406fc2adb23cdf17a1aaf5744ac is the first bad commit
commit 769fb1547fc50406fc2adb23cdf17a1aaf5744ac
Author: Lucio Franco <luciofranco14@gmail.com>
Date:   Thu Feb 24 13:39:37 2022 -0500

    tokio: Add initial io driver metrics (#4507)

 .gitignore                           |  2 +
 tests-integration/Cargo.toml         |  2 +-
 tokio/src/io/driver/metrics.rs       | 22 +++++++++
 tokio/src/io/driver/mod.rs           | 38 +++++++++++++++-
 tokio/src/lib.rs                     |  2 +-
 tokio/src/macros/cfg.rs              |  6 +++
 tokio/src/runtime/metrics/io.rs      | 30 +++++++++++++
 tokio/src/runtime/metrics/mod.rs     |  5 +++
 tokio/src/runtime/metrics/runtime.rs | 87 ++++++++++++++++++++++++++++++++++++
 tokio/src/runtime/mod.rs             |  4 ++
 tokio/tests/rt_metrics.rs            | 34 ++++++++++++++
 11 files changed, 229 insertions(+), 3 deletions(-)
 create mode 100644 tokio/src/io/driver/metrics.rs
 create mode 100644 tokio/src/runtime/metrics/io.rs
 ```
 
Compilation is only successful with the `net` feature.

--- a/tokio/src/io/driver/metrics.rs
+++ b/tokio/src/io/driver/metrics.rs
@@ -4,7 +4,7 @@
 //! these need to be available in the case when `net` is enabled but
 //! `rt` is not.
 
-cfg_not_rt_and_metrics! {
+cfg_not_rt_and_metrics_and_net! {
     #[derive(Default)]
     pub(crate) struct IoDriverMetrics {}
 
@@ -15,8 +15,10 @@ cfg_not_rt_and_metrics! {
     }
 }
 
-cfg_rt! {
-    cfg_metrics! {
-        pub(crate) use crate::runtime::IoDriverMetrics;
+cfg_net! {
+    cfg_rt! {
+        cfg_metrics! {
+            pub(crate) use crate::runtime::IoDriverMetrics;
+        }
     }
 }

--- a/tokio/src/io/driver/mod.rs
+++ b/tokio/src/io/driver/mod.rs
@@ -290,19 +290,21 @@ cfg_not_rt! {
     }
 }
 
-cfg_metrics! {
-   impl Handle {
-        // TODO: Remove this when handle contains `Arc<Inner>` so that we can return
-        // &IoDriverMetrics instead of using a closure.
-        //
-        // Related issue: https://github.com/tokio-rs/tokio/issues/4509
-        pub(crate) fn with_io_driver_metrics<F, R>(&self, f: F) -> Option<R>
-        where
-            F: Fn(&IoDriverMetrics) -> R,
-        {
-            self.inner().map(|inner| f(&inner.metrics))
-        }
-   }
+cfg_net! {
+    cfg_metrics! {
+       impl Handle {
+            // TODO: Remove this when handle contains `Arc<Inner>` so that we can return
+            // &IoDriverMetrics instead of using a closure.
+            //
+            // Related issue: https://github.com/tokio-rs/tokio/issues/4509
+            pub(crate) fn with_io_driver_metrics<F, R>(&self, f: F) -> Option<R>
+            where
+                F: Fn(&IoDriverMetrics) -> R,
+            {
+                self.inner().map(|inner| f(&inner.metrics))
+            }
+       }
+    }
 }
 
 impl Handle {

--- a/tokio/src/macros/cfg.rs
+++ b/tokio/src/macros/cfg.rs
@@ -195,9 +195,19 @@ macro_rules! cfg_not_metrics {
     }
 }
 
-macro_rules! cfg_not_rt_and_metrics {
+macro_rules! cfg_not_rt_and_metrics_and_net {
     ($($item:item)*) => {
-        $( #[cfg(not(all(feature = "rt", all(tokio_unstable, not(loom)))))] $item )*
+        $( #[cfg(not(all(feature = "net", feature = "rt", all(tokio_unstable, not(loom)))))]$item )*
+    }
+}
+
+macro_rules! cfg_net_or_process {
+    ($($item:item)*) => {
+        $(
+            #[cfg(any(feature = "net", feature = "process"))]
+            #[cfg_attr(docsrs, doc(cfg(any(feature = "net", feature = "process"))))]
+            $item
+        )*
     }
 }
 

